### ISSUE TITLE
why loss is 0???

### ISSUE THREAD

**Issue by HHL-445** [lanyunshi/ConversationalKBQA#10]
I can access query.wikidata but why can't I count the loss. The model runs very fast, has anyone run it?
dev reward=(0.0, 0.0) 
dev te reward=(0.0, 0.0) 
test reward=(0.0, 0.0) 
training loss=0.0 
training reward=(0.0, 0.0) 
training te loss=0.0 
training te reward=(0.0, 0.0)

**Comment by iCloud112** [comment]
I had the same problem before, but I don't know why it worked later without any changes.

**Comment by HHL-445** [comment]
> 

I suspect that the not_update parameter of the retrieve_via_frontier function is wrong. He should be False. Otherwise, the subgraph cannot be retrieved. But I still run very slowly

**Comment by iCloud112** [comment]
Are you accessing the url "https://query.wikidata.org/sparql" with the help of vpn？I also don't know why. I am also trying to train a new model but I think there are some problems with the model I trained. The code provided by the author is a bit messy and has some errors.

**Comment by HHL-445** [comment]
yes.I use VPN .
If it's convenient, can you add a contact information for communication and discussion（）
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lanyunshi/ConversationalKBQA"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月5日(星期四) 下午5:36
***@***.***&gt;;
抄送:&nbsp;"西南石油大学 ***@***.******@***.***&gt;;
主题:&nbsp;Re: [lanyunshi/ConversationalKBQA] why loss is 0??? (Issue #10)





 
Are you accessing the url "https://query.wikidata.org/sparql" with the help of vpn？I also don't know why. I am also trying to train a new model but I think there are some problems with the model I trained. The code provided by the author is a bit messy and has some errors.

**Comment by HHL-445** [comment]
Yes I use VPN.If you have any new progress, you can share it with me. Thanks

**Comment by iCloud112** [comment]
You too. Thank you!

**Comment by Bigchen8013** [comment]
> 



> I can access query.wikidata but why can't I count the loss. The model runs very fast, has anyone run it? dev reward=(0.0, 0.0) dev te reward=(0.0, 0.0) test reward=(0.0, 0.0) training loss=0.0 training reward=(0.0, 0.0) training te loss=0.0 training te reward=(0.0, 0.0)

Hi
I have the same problem, have you solved it? 
If you have solved it, can you share with me？
Thank you！

**Comment by HHL-445** [comment]
@Bigchen8013 The value of not_update in the retrieve_KB function is args.do_train. The retrieve_via_frontier function is called in the retrieve_KB function, and the value of not_update is also passed. But the value of arg.do_train is 1 during training and the value that not_update needs to pass in retrieve_via_frontier is False. I can successfully calculate the loss after adding not_update = False to the front of retrieve_via_frontier. You can try
